src/main/java/org/myrobotlab/service/FiniteStateMachine.java

```diff
@@ -58,6 +63,21 @@ public class Tuple {
     public Transition transition;
     public StateTransition stateTransition;
   }
+  
+  public class StateChange {

```

Comment:
Probably should have javadoc for all of this, and use getters/setters instead of public fields